packages/nitro-protocol/src/index.ts

```diff
@@ -1,29 +1,46 @@
-import NitroAdjudicatorArtifact from '../artifacts/contracts/NitroAdjudicator.sol/NitroAdjudicator.json';
-import TrivialAppArtifact from '../artifacts/contracts/TrivialApp.sol/TrivialApp.json';
-import TokenArtifact from '../artifacts/contracts/Token.sol/Token.json';
-import AssetHolderArtifact from '../artifacts/contracts/AssetHolder.sol/AssetHolder.json';
-import Erc20AssetHolderArtifact from '../artifacts/contracts/ERC20AssetHolder.sol/ERC20AssetHolder.json';
-import EthAssetHolderArtifact from '../artifacts/contracts/ETHAssetHolder.sol/ETHAssetHolder.json';
-import TestNitroAdjudicatorArtifact from '../artifacts/contracts/test/TESTNitroAdjudicator.sol/TESTNitroAdjudicator.json';
-import TestAssetHolderArtifact from '../artifacts/contracts/test/TESTAssetHolder.sol/TESTAssetHolder.json';
+import pick from 'lodash.pick';
+
+import FULLTokenArtifact from '../artifacts/contracts/Token.sol/Token.json';
+import FULLNitroAdjudicatorArtifact from '../artifacts/contracts/NitroAdjudicator.sol/NitroAdjudicator.json';
+import FULLErc20AssetHolderArtifact from '../artifacts/contracts/ERC20AssetHolder.sol/ERC20AssetHolder.json';
+import FULLEthAssetHolderArtifact from '../artifacts/contracts/ETHAssetHolder.sol/ETHAssetHolder.json';
+import FULLTestNitroAdjudicatorArtifact from '../artifacts/contracts/test/TESTNitroAdjudicator.sol/TESTNitroAdjudicator.json';
+import FULLTestAssetHolderArtifact from '../artifacts/contracts/test/TESTAssetHolder.sol/TESTAssetHolder.json';
+
+// https://hardhat.org/guides/compile-contracts.html#artifacts
+const fields = [
+  'contractName',
+  'abi',
+  'bytecode',
+  'deployedByteCode',
+  'linkReferences',
+  'deployedLinkReferences',
+];
+
+interface MinimalArtifact {
+  contractName: typeof FULLNitroAdjudicatorArtifact.contractName;
```

Comment:
What is `typeof FULLNitroAdjudicatorArtifact`? I don't understand how it could be a meaningful type, if `FULLNitroAdjudicatorArtifact` is an imported `json` file.

Reply:
There is some inference done by the typescript compiler, such that importing from a Json file results in a rich and typed object. I was surprised by this, too.

Reply:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-9.html#new---resolvejsonmodule